coordinator/src/main/scala/filodb.coordinator/queryplanner/LongTimeRangePlanner.scala

```diff
@@ -1,22 +1,50 @@
 package filodb.coordinator.queryplanner
 
-import filodb.query.{LogicalPlan, QueryContext}
-import filodb.query.exec.ExecPlan
+import filodb.coordinator.queryplanner.LogicalPlanUtils._
+import filodb.query.{LogicalPlan, PeriodicSeriesPlan, QueryContext}
+import filodb.query.exec.{ExecPlan, PlanDispatcher, StitchRvsExec}
 
 /**
   * LongTimeRangePlanner knows about limited retention of raw data, and existence of downsampled data.
   * For any query that arrives beyond the retention period of raw data, it splits the query into
   * two time ranges - latest time range from raw data, and old time range from downsampled data.
+  * It then stitches the subquery results to present top level query results for entire time range.
   *
   * @param rawClusterPlanner this planner (typically a SingleClusterPlanner) abstracts planning for raw cluster data
   * @param downsampleClusterPlanner this planner (typically a SingleClusterPlanner)
   *                                 abstracts planning for downsample cluster data
   */
 class LongTimeRangePlanner(rawClusterPlanner: QueryPlanner,
-                           downsampleClusterPlanner: QueryPlanner) extends QueryPlanner {
+                           downsampleClusterPlanner: QueryPlanner,
+                           rawDataRetentionMillis: Long,
+                           stitchDispatcher: => PlanDispatcher) extends QueryPlanner {
 
   def materialize(logicalPlan: LogicalPlan, qContext: QueryContext): ExecPlan = {
-    ???
-  }
+    logicalPlan match {
+      case p: PeriodicSeriesPlan =>
+        val lastRawTime = System.currentTimeMillis() - rawDataRetentionMillis
```

Comment:
Using the wall clock here seems very very wrong. I would assume that the stitching would operate by finding the stitch point from the actual data, not by an externally provided "retention" timestamp.

Reply:
Good point - this first implementation solves the real-time ingested data. Moreover we downsample real-time data only - i.e. data where ingestion and user time is approx the same. 

We have to deal with use cases with historical data ingested separately. As much as I would like to solve this use case right away, it is a bigger discussion that deserves separate treatment and I don't want to include that in the scope of this PR.